docs/content/docs/tutorials/ec2-example.md

```diff
@@ -0,0 +1,491 @@
+ 
+ 
+### Create a VPC Workflow
+ 
+This section details the steps to create a network topology consisting of multiple, connected resources from a single manifest file. The following resources will be present in said manifest:
+* 1 VPC
+* 1 Instance
+* 1 Internet Gateway
+* 1 NAT Gateways
+* 1 Elastic IPs
+* 2 Route Tables
+* 2 Subnets (1 Public; 1 Private)
+* 1 Security Group
+ 
+ 
+The VPC is connected to the internet through an Internet Gateway. The NAT Gateway is created in the public Subnet with an associated Elastic IP. The Instance is deployed into the private Subnet which can connect to the internet using the NAT Gateway in the public Subnet. Lastly, one Route Table (public) will contain a route to the Internet Gateway while the other Route Table (private) contains a route to the NAT Gateway.
+
+{{% hint type="info" title="Referencing Resources" %}}
+Notice that the ACK custom resources reference each other using "*Ref" fields inside the manifest and the user does not have to worry about finding `vpc-ID` when creating the Subnet resource manifests.
+ 
+Refer to [API Reference](https://aws-controllers-k8s.github.io/community/reference/) for *EC2*
+to find the supported reference fields.
+{{% /hint %}}
+
+* Deploy the resources using the provided YAML and `kubectl apply -f vpc-workflow.yaml`:
+ 
+```
+apiVersion: ec2.services.k8s.aws/v1alpha1
+kind: VPC
+metadata:
+  name: tutorial-vpc
+spec:
+  cidrBlocks: 
+  - 10.0.0.0/16
+  enableDNSSupport: true
+  enableDNSHostnames: true
+  tags:
+    - key: name
+      value: vpc-tutorial
+---
+apiVersion: ec2.services.k8s.aws/v1alpha1
+kind: InternetGateway
+metadata:
+  name: tutorial-igw
+spec:
+  vpcRef:
+    from:
+      name: tutorial-vpc
+---
+apiVersion: ec2.services.k8s.aws/v1alpha1
+kind: NATGateway
+metadata:
+  name: tutorial-natgateway1
+spec:
+  subnetRef:
+    from:
+      name: tutorial-public-subnet1
+  allocationRef:
+    from:
+      name: tutorial-eip1
+---
+apiVersion: ec2.services.k8s.aws/v1alpha1
+kind: ElasticIPAddress
+metadata:
+  name: tutorial-eip1
+spec:
+  tags:
+    - key: name
+      value: eip-tutorial
+---
+apiVersion: ec2.services.k8s.aws/v1alpha1
+kind: RouteTable
+metadata:
+  name: tutorial-public-route-table
+spec:
+  vpcRef:
+    from:
+      name: tutorial-vpc
+  routes:
+  - destinationCIDRBlock: 0.0.0.0/0
+    gatewayRef:
+      from:
+        name: tutorial-igw
+---
+apiVersion: ec2.services.k8s.aws/v1alpha1
+kind: RouteTable
+metadata:
+  name: tutorial-private-route-table-az1
+spec:
+  vpcRef:
+    from:
+      name: tutorial-vpc
+  routes:
+  - destinationCIDRBlock: 0.0.0.0/0
+    natGatewayRef:
+      from:
+        name: tutorial-natgateway1
+---
+apiVersion: ec2.services.k8s.aws/v1alpha1
+kind: Subnet
+metadata:
+  name: tutorial-public-subnet1
+spec:
+  availabilityZone: us-west-2a
```

Comment:
May want to make a note that the user needs to change the AZ if they're not using `us-west-2` region. My subnets originally said "created" with no errors, but I was not able to find them in the AWS Console because the VPC was in `us-east-2`:

```
vpc.ec2.services.k8s.aws/tutorial-vpc created
...
subnet.ec2.services.k8s.aws/tutorial-public-subnet1 created
subnet.ec2.services.k8s.aws/tutorial-private-subnet1 created
```

Once I changed the AZs in this file, they created successfully.

```
vpc.ec2.services.k8s.aws/tutorial-vpc unchanged
...
subnet.ec2.services.k8s.aws/tutorial-public-subnet1 configured
subnet.ec2.services.k8s.aws/tutorial-private-subnet1 configured
```

Related question: Why did the subnet report successful creation even though it didn't actually get created?

Reply:
Discussed offline re: subnet creation. This message means that the Kubernetes object has been created, but it will take some time to reconcile that with the actual AWS resources (especially for a subnet, which has dependencies on the NAT gateway and route table that must be created first).